### PR TITLE
Version bump to libhdhomerun-20190621

### PR DESCRIPTION
**Pull request by thebitpit**
Upgrade to EAPI 7
Add myself to metadata

Signed-off-by: Wilson Michaels <thebitpit@eartlink.net>

### CODE REVIEW

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit eutils
```
Comment (mgorny) [inline media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Where are you using eutils?

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${PN}"
+
+src_prepare() {
```
Comment (mgorny) [inline media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Why do you need that?

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	default
+}
+
+src_configure() {
```
Comment (mgorny) [inline media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Why do you need that? A comment in the ebuild would be helpful.

File: media-libs/libhdhomerun/metadata.xml [hunk 4]
```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-<!-- maintainer-needed -->
+	<maintainer type="person">
+		<email>thebitpit@earthlink.net</email>
+		<name>Wilson Michaels</name>
+		<description>Proxy maintainer</description>
```
Comment (mgorny) [inline media-libs/libhdhomerun/metadata.xml]: We don't do that for a long time now.

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild (hunk 1 above)
Comment (thebitpit) [reply media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: nowhere - remove

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild (hunk 2 above)
Comment (thebitpit) [reply media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Remove not needed

File: media-libs/libhdhomerun/metadata.xml (hunk 4 above)
Comment (thebitpit) [reply media-libs/libhdhomerun/metadata.xml]: Fixed

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild (hunk 3 above)
Comment (thebitpit) [reply media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Remove not needed

**Comment by thebitpit** [comment]
I addressed requested changes.
The PR is updated.

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+PATCHES=(
+	"${FILESDIR}"/dont-strip.patch
+)
+
+LICENSE="LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND=""
```
Comment (mgorny) [inline media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: It's up to you to make the final decision but I'd suggest you removed those empty variables. It looks weird that you assign an empty variable to an empty variable ;-).

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="SiliconDust HDHomeRun Utilties"
+HOMEPAGE="https://www.silicondust.com/support/linux/"
+SRC_URI="https://download.silicondust.com/hdhomerun/${PN}_${PV}.tgz"
+
+PATCHES=(
```
Comment (mgorny) [inline media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Generic suggestion is to keep PATCHES below the basic variables, i.e. below where  you keep RDEPEND right now.

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild (hunk 5 above)
Comment (thebitpit) [reply media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Good comment - removed

File: media-libs/libhdhomerun/libhdhomerun-20190621.ebuild (hunk 6 above)
Comment (thebitpit) [reply media-libs/libhdhomerun/libhdhomerun-20190621.ebuild]: Moved PATCHES

**Comment by thebitpit** [comment]
Fix style comments.

**Comment by mgorny** [comment]
Ok, I'm going to have a few more issues/suggestions.

1. You should be able to replace no-stripping-patch with passing `STRIP=:` to `emake`. This implies defining `src_compile()` but should be less intrusive than carrying a patch.
2. It calls `gcc` instead of `${CC}`. This should be just a matter of passing `CC="$(tc-getCC)"` to `emake` to override the Makefile value.
3. The package overrides `-O` level by appending `-O2`. We really don't want it to do that as it doesn't let user change optimization level.
4. It would be nice to split compilation into separately compiling every `.c` file. This will make the package much happier with parallel make, distcc, warning/error reporting, etc. However, this is major Makefile change, probably worth asking upstream for.
5. It would also be nice to make `hdhomerun_config` link to the shared library instead of compiling all `.c` files twice.

**Comment by thebitpit** [comment]
I tried item 1. STRIP=: and it does not work. The Makefile sets the STRIP variable ignoring the existing value. Only the patch works.
All of the items require patching the Makefile,
Given that no one has complained for years it is probably not used by many people. (I own 2 devices and use them for hours everyday.)

**Comment by mgorny** [comment]
Impossible! Passing them as arguments to emake (not envvars) should override the values

**Comment by thebitpit** [comment]
OK. I'll try again. I must be doing something wrong.

**Comment by thebitpit** [comment]
The package has no `configuration` file. I think a single `Makefile` figures out the OS being used, sets the bin files' extensions, then builds the right kind of binaries. There is a file `hdhomerun_config.c` to configure the hardware.

Does `emake` modify a copy of the supplied `Makefile` and use that?

**Comment by mgorny** [comment]
> The package has no `configuration` file. I think a single `Makefile` figures out the OS being used, sets the bin files' extensions, then builds the right kind of binaries. There is a file `hdhomerun_config.c` to configure the hardware.
> 
> Does `emake` modify a copy of the supplied `Makefile` and use that?

No, it's just a thin wrapper over make. Make does the right thing.

    emake CC=$(tc-getCC) STRIP=:

should do what you need. At least in `src_compile`. You may also need to pass it in `src_install` but it shouldn't be necessary if it doesn' recompile the files (it shouldn't).

**Comment by thebitpit** [comment]
I ran with out patches and included:
```
src_install() {
        emake CC=$(tc-getCC) STRIP=:
}
```
and got:
```
* QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/hdhomerun_config
 * QA Notice: Pre-stripped files found:
 * /usr/bin/hdhomerun_config

```
At the top of the `Makefile` they are set like this:
`CC    := $(CROSS_COMPILE)gcc`
`STRIP := $(CROSS_COMPILE)strip`

**Comment by mgorny** [comment]
Did you include `src_compile`?

**Comment by thebitpit** [comment]
I goofed and used `src_install()` instead of `src_compile()`.
I added:
`inherit toolchain-funcs`
and it works! It fixes the first two items:
 >     1. You should be able to replace no-stripping-patch with passing `STRIP=:` to `emake`. This implies defining `src_compile()` but should be less intrusive than carrying a patch.
> 
>     2. It calls `gcc` instead of `${CC}`. This should be just a matter of passing `CC="$(tc-getCC)"` to `emake` to override the Makefile value.
> 

Is it best preactice to make a `PATCH` or create a `sed...` to remove  `-O2`?
>     3. The package overrides `-O` level by appending `-O2`. We really don't want it to do that as it doesn't let user change optimization level.

**Comment by mgorny** [comment]
> Is it best preactice to make a `PATCH` or create a `sed...` to remove `-O2`?

Depends. If simple sed is sufficient and looks reliably, you can use it. If patch looks more reliably and won't require frequent updates, you can use it. Probably sed in this case.

**Comment by thebitpit** [comment]
> 
>     4. It would be nice to split compilation into separately compiling every `.c` file. This will make the package much happier with parallel make, distcc, warning/error reporting, etc. However, this is major Makefile change, probably worth asking upstream for.
This is a small package. I'll defer this.
> 
>     5. It would also be nice to make `hdhomerun_config` link to the shared library instead of compiling all `.c` files twice.
Fixed with a patch.

**Comment by mgorny** [comment]
Looks good, thanks!

**Comment by thebitpit** [comment]
> Also, please look at other commit messages in the repository and follow suit.

Changed the PR to:
`media-libs/libhdhomerun: Bump to version 20190621 bug #676140`

Changed the commit message to:
```
media-libs/libhdhomerun: Bump to version 20190621 bug #676140

Upgrade to EAPI 7

Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: Wilson Michaels <thebitpit@eartlink.net>
Is more detail required? Do I need to make a new pull request?
```
Is there anything else?

**Pull request closed without merging**